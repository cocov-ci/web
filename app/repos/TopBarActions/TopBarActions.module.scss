.searchField {
  margin-right: 16px;
}

.topBarActions {
  display: flex;
  align-items: center;
}
