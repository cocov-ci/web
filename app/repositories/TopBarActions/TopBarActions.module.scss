.searchField {
  margin-right: 16px;
}
