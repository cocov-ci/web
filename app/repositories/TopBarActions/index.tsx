'use client'

import Button from 'app/common/Button'
import SearchField from 'app/common/SearchField'

import styles from './TopBarActions.module.scss'

interface TopBarActionsProps {
  searchBarDisabled?: boolean
  searchBarLoading?: boolean
  onSearchChange: (arg: string) => void
}

const TopBarActions = ({
  searchBarLoading = false,
  onSearchChange,
}: TopBarActionsProps) => {
  return (
    <div className={styles.topBarActions}>
      <SearchField
        className={styles.searchField}
        loading={searchBarLoading}
        onSearch={onSearchChange}
      />
      <Button>Add Repository</Button>
    </div>
  )
}

export default TopBarActions
