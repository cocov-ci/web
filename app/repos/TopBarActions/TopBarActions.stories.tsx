import type { Meta, StoryObj } from '@storybook/react'

import TopBarActions from './index'

const meta: Meta<typeof TopBarActions> = {
  title: 'Repositories/TopBarActions',
  component: TopBarActions,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof TopBarActions>

export const Default: Story = {
  args: {
    searchFieldLoading: false,
    searchFieldDisabled: false,
    onSearchChange: () => null,
  },
}

export const SearchFieldLoading: Story = {
  args: {
    searchFieldLoading: true,
    searchFieldDisabled: false,
    onSearchChange: () => null,
  },
}

export const SearchFieldDisabled: Story = {
  args: {
    searchFieldLoading: false,
    searchFieldDisabled: true,
    onSearchChange: () => null,
  },
}
