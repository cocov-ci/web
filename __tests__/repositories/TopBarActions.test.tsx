import { render, screen } from '@testing-library/react'

import TopBarActions from 'app/repos/TopBarActions'

import '@testing-library/jest-dom'

describe('Repositories/TopBarActions', () => {
  const { container } = render(<TopBarActions onSearchChange={() => null} />)

  it('renders TopBarActions component', () => {
    expect(screen.getByPlaceholderText('Type to Search')).toBeVisible()
    expect(screen.getByRole('button')).toHaveTextContent('Add Repository')
    expect(container.querySelector('svg')).toBeVisible()
  })

  it('renders TopBarActions component with searchFieldLoading props', () => {
    const { container } = render(
      <TopBarActions onSearchChange={() => null} searchFieldLoading={true} />,
    )

    expect(screen.getByPlaceholderText('Type to Search')).toBeVisible()
    expect(screen.getByRole('button')).toHaveTextContent('Add Repository')
    expect(container.querySelector('svg')).toBeVisible()
    expect(container.querySelector('img')).toBeVisible()
  })

  it('renders TopBarActions component with searchFieldDisabled props', () => {
    const { container } = render(
      <TopBarActions onSearchChange={() => null} searchFieldDisabled={true} />,
    )

    expect(screen.getByPlaceholderText('Type to Search')).toBeVisible()
    expect(screen.getByRole('button')).toHaveTextContent('Add Repository')
    expect(container.querySelector('svg')).toBeVisible()
    expect(container.querySelector('input')).toHaveAttribute('disabled')
  })

  it('renders TopBarActions snapshots', () => {
    expect(container).toMatchSnapshot()
  })
})
