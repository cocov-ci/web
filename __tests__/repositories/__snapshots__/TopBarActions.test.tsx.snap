// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories/TopBarActions renders TopBarActions snapshots 1`] = `<div />`;
